Docs: Add recommendations for CSS import order (#64321)
vercel/next.js: docs/02-app/01-building-your-application/05-styling/01-css-modules.mdx

</PagesOnly>

> CSS Modules are **only enabled for files with the `.module.css` extension**.

Author: ```suggestion
CSS Modules are **only enabled for files with the `.module.css` and `.module.sass` extensions**.
```

```tsx filename="base-button.tsx" switcher
import styles from './base-button.module.css'

export function BaseButton() {
  return <button className={styles.primary} />
}
```

```jsx filename="base-button.js" switcher
import styles from './base-button.module.css'

export function BaseButton() {
  return <button className={styles.primary} />
}
```

```tsx filename="page.ts" switcher
import { BaseButton } from './base-button'
import styles from './page.module.css'

export function Page() {
  return <BaseButton className={styles.primary} />
}
```

```jsx filename="page.js" switcher
import { BaseButton } from './base-button'
import styles from './page.module.css'

export function Page() {
  return <BaseButton className={styles.primary} />
}
```

To maintain a predictable order, we recommend the following:

- Only import a CSS file in a single JS/TS file.
  - If using global class names, import the global styles in the same file in the order you want them to be applied.
- Prefer CSS Modules over global styles.
  - Use a consistent naming convention for your CSS modules. For example, using `<name>.module.css` over `<name>.tsx`.

> - Extract shared styles into a separate file and import them into the files that need them.

Author: ```suggestion
- Extract shared styles into a separate shared component and use composition. 
```

> - If using [Tailwind](/docs/app/building-your-application/styling/tailwind-css), import the stylesheets at the top of the file, preferably in the [Root Layout](/docs/app/building-your-application/routing/pages-and-layouts#root-layout-required).

Author: ```suggestion
- If using [Tailwind](/docs/app/building-your-application/styling/tailwind-css), import the stylesheet at the top of the file, preferably in the [Root Layout](/docs/app/building-your-application/routing/pages-and-layouts#root-layout-required).
```
